### DISCUSSION TITLE
crossfade

### DISCUSSION THREAD

**Discussion by MYAzab** [google/filament#4228] [General]
is there away to **crossfade** two glTF animations?

**Comment by prideout** [comment]
Yes, you can have many concurrent animations, each with their own time parameter.  Simply call `applyAnimation` separately for each `<animation>`.

**Reply by MYAzab** [reply]
Thanks dear @prideout,

i just want to make sure we understand each other.., and this feature is really important when it comes to move from action to another smoothly..

say we have a skinned .glb character,.. with two clips inside, now, the crossfading plays the role of (blending-with-overlapping) the stored values of the transformation data (as keyframes for translate, rotate and _scale_) between both the end-of-the-first-clip and start-of-the-sec-clip, while the duration of morphing time depends on that intersection,..

in a 3d software we do the crossfading visually by overlapping two clips (on the same track) with specific number of frames (which considered the morphing time "crossfading time" between the the end of first clip, with the start of the second clip) as i mentioned; technically, it's fade-out and fade-in crossed with each others or overlapped over time, 

so, does the `applyAnimation` do that?..

you may have a look at:
https://threejs.org/docs/#api/en/animation/AnimationAction

thanks in advance

**Reply by prideout** [reply]
Well, the `Animator` interface (which is what proffers `applyAnimation`) is a low-level stateless API for applying the individual animations that are defined in the asset.  It is not a stateful interface like three's AnimationAction, which has functions like "play", "stop", "fadeIn".  See also #3121.

**Reply by MYAzab** [reply]
thank you @prideout,
i will try to play around it..

**Reply by XperienceLabs** [reply]
is there any solution available yet!
this is result getting in filament gltf viewer
[Screencast from 10-06-23 05:13:08 PM IST.webm](https://github.com/google/filament/assets/121343777/4306f04c-4d2e-4045-86a1-7c6267ff1cc6)
